src/components/Counter/Counter.test.tsx

```diff
@@ -0,0 +1,30 @@
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Counter from './Counter';
+
+describe('Counter', () => {
+    it('should render with initial value', () => {
+        const { getByText } = render(<Counter initialValue={0} />);
+        const expectedText = /Counter: 0/i;
+        expect(getByText(expectedText)).toBeInTheDocument();
+    });
+
+    it('should increment value when `increase` button pressed', () => {
+        const { getByText } = render(<Counter initialValue={0} />);
+        act(() => {
+            userEvent.click(screen.getByText('Increase'));
+        });
```

Comment:
I am not sure that you need act.
Please check fireEvents from ract testing library.